Python/bytecodes.c

```diff
@@ -3860,6 +3878,13 @@ dummy_func(
                     assert(PyTuple_CheckExact(attr));
                     assert(func_obj->func_defaults == NULL);
                     func_obj->func_defaults = attr;
+                    PyCodeObject *code = (PyCodeObject *)func_obj->func_code;
+                    if (code->co_expected_number_of_defaults < 0) {
+                        code->co_expected_number_of_defaults = (int32_t)PyTuple_GET_SIZE(attr);
+                    }
+                    else if (code->co_expected_number_of_defaults != PyTuple_GET_SIZE(attr)) {
+                        func_obj->func_version = 0;
+                    }
```

Comment:
This works, but couldn't we let the bytecode compiler give us the number? Again, I worry a bit about the size of the JIT code.